### ISSUE TITLE
Could not load library `edfapi`

### ISSUE THREAD

**Issue by likanzhan** [grero/Eyelink.jl#16]
Hi Roger, 
I don't know when this begins to occur.
But I now encountered an 'could not load library' error, when I tries to use the library:

```
julia> Eyelink.version()
ERROR: could not load library "/Library/Frameworks/edfapi.framework/Versions/Current/edfapi"
dlopen(/Library/Frameworks/edfapi.framework/Versions/Current/edfapi.dylib, 0x0001): tried: '/Library/Frameworks/edfapi.framework/Versions/Current/edfapi.dylib' (no such file), '/Library/Frameworks/edfapi.framework/Versions/A/edfapi.dylib' (no such file)
Stacktrace:
 [1] version()
   @ Eyelink ~/.julia/packages/Eyelink/6MnRj/src/Eyelink.jl:18
 [2] top-level scope
   @ REPL[25]:1
```

Any suggestions?

Thanks,

The version of Julia I used is
```
julia> versioninfo()
Julia Version 1.8.1
Commit afb6c60d69a (2022-09-06 15:09 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.5.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
Environment:
  JULIA_NUM_THREADS = 1
  JULIA_PKG_USE_CLI_GIT = true
```

**Comment by grero** [comment]
That is weird. For fear of stating the obvious, have you verified that the edf library actually still exists at the location mentioned in the error?

**Comment by likanzhan** [comment]
@grero Yes, it actually exists there. One possible reason is that I have updated the `XCode` since then. Could this be a cause?

**Comment by likanzhan** [comment]
I have checked, the library is there. but no `edfapi.dylib` is found there:

<img width="224" alt="Screen Shot 2022-10-13 at 22 59 25" src="https://user-images.githubusercontent.com/10531698/195633239-36f54398-224c-4ac2-8048-85cf41122724.png">

**Comment by likanzhan** [comment]
@grero

But as it is described in page 24 of the file `EDF Access C API user manual`, the missing of the `dylib` should no be the problem.

<img width="1348" alt="Screen Shot 2022-10-19 at 13 09 31" src="https://user-images.githubusercontent.com/10531698/196604054-584d5f1b-2862-4cd7-b21f-b3b8f5ea7da5.png">

**Comment by grero** [comment]
Oh I see, that would be a problem then. I guess when compiling c-code, you only need the headers. However, the way I wrote the julia code, you do need the dylib. 
Could you perhaps try downloading an older version which does contain the dylib file?
In the long run, it seems like I'd have to find a way to not depend on the dylib, though.

**Comment by likanzhan** [comment]
@grero 

This error turns out to be related to the Julia version I used.
The edfapi is built for X86 only.
So if I switch back to the `macOS x86 (Intel or Rosetta)`, the package worked successfully as expected,
but if I use the `macOS ARM (M-series Processor) version`, the error will be reported.

It seems that the edfapi cannot be compiled in a macOS ARM system.

Thanks again.
